src/pp.js

```diff
@@ -360,7 +367,13 @@ function print(options, doc) {
 
                 pos = 0;
               } else {
-                out.push("\n" + _repeatIndent(ind, indentStr));
+                let indentLine;
+                if (ind in indentCache) {
+                  indentLine = indentCache[ind];
+                } else {
+                  indentCache[ind] = indentLine = _repeatIndent(ind, indentStr);
+                }
+                out.push("\n" + indentLine);
```

Comment:
Can you run your benchmark and see if a function call makes any difference? If you are going this route, may as drop the caching from the indent function altogether since you are doing it here.